service/k8sclient/operations/cluster_create.go

```diff
@@ -0,0 +1,192 @@
+)
+
+// NewClusterCreate returns new object of ClusterCreate.
+func NewClusterCreate(l logger.Logger, name string, size int32) *ClusterCreate {
+	return &ClusterCreate{
+		kubectl:  kubectl.NewKubeCtl(l),
+		l:        l,
+		name:     name,
+		size:     size,
+		pxcImage: "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0",
+		kind:     "PerconaXtraDBCluster",
+	}
+}
+
+// ClusterCreate creates new kubernetes cluster.
+type ClusterCreate struct {
+	kubectl *kubectl.KubeCtl
+	l       logger.Logger
+
+	name     string
+	size     int32
+	pxcImage string
+	kind     string
+}
+
+// Start starts new cluster creating process.
+func (c *ClusterCreate) Start(ctx context.Context) error {
+	res := &pxc.PerconaXtraDBCluster{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "pxc.percona.com/v1-4-0",
+			Kind:       c.kind,
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: pxc.PerconaXtraDBClusterSpec{
+			AllowUnsafeConfig: true,
+			SecretsName:       "my-cluster-secrets",
+
+			PXC: &pxc.PodSpec{
+				Size:  c.size,
+				Image: c.pxcImage,
+				VolumeSpec: &pxc.VolumeSpec{
+					PersistentVolumeClaim: &core.PersistentVolumeClaimSpec{
+						Resources: core.ResourceRequirements{
+							Requests: core.ResourceList{
+								core.ResourceStorage: resource.MustParse("1Gi"),
+							},
+						},
+					},
+				},
+				Affinity: &pxc.PodAffinity{
+					TopologyKey: pointer.ToString(pxc.AffinityTopologyKeyOff),
+				},
+			},
+
+			ProxySQL: &pxc.PodSpec{
+				Enabled: true,
+				Size:    c.size,
+				Image:   "percona/percona-xtradb-cluster-operator:1.4.0-proxysql",
+				VolumeSpec: &pxc.VolumeSpec{
+					PersistentVolumeClaim: &core.PersistentVolumeClaimSpec{
+						Resources: core.ResourceRequirements{
+							Requests: core.ResourceList{
+								core.ResourceStorage: resource.MustParse("1Gi"),
+							},
+						},
+					},
+				},
+				Affinity: &pxc.PodAffinity{
+					TopologyKey: pointer.ToString(pxc.AffinityTopologyKeyOff),
+				},
+			},
+
+			PMM: &pxc.PMMSpec{
+				Enabled: false,
+			},
+
+			//Backup: &pxc.PXCScheduledBackup{
+			//	Image: backupImage,
+			//	Schedule: []pxc.PXCScheduledBackupSchedule{{
+			//		Name:        "test",
+			//		Schedule:    "*/1 * * * *",
+			//		Keep:        3,
+			//		StorageName: backupStorageName,
+			//	}},
+			//	Storages: map[string]*pxc.BackupStorageSpec{
+			//		backupStorageName: {
+			//			Type: pxc.BackupStorageFilesystem,
+			//			Volume: &pxc.VolumeSpec{
+			//				PersistentVolumeClaim: &core.PersistentVolumeClaimSpec{
+			//					Resources: core.ResourceRequirements{
+			//						Requests: core.ResourceList{
+			//							core.ResourceStorage: resource.MustParse("1Gi"),
+			//						},
+			//					},
+			//				},
+			//			},
+			//		},
+			//	},
+			//	ServiceAccountName: "percona-xtradb-cluster-operator",
+			//},
+		},
+	}
+	return c.apply(ctx, res)
+}
+
+// Wait waits until cluster is ready.
+func (c *ClusterCreate) Wait(ctx context.Context) error {
```

Comment:
> Given that, in most cases, operation status is equal to CR status.state, and a number of special cases is small, dbaas-controller's Kubernetes client will not expose methods for waiting for an operation to complete

Please re-read the whole Confluence page – it changed significantly since it mentioned Wait methods